articles/102_dependency_groups.md

```diff
@@ -0,0 +1,168 @@
+---
+layout: default
+title: Dependency groups
+permalink: articles/dependency_groups.html
+abstract:
+  This article describes the concept of dependency groups, their use cases, how they can described in the package manifest, and how they can be mapped into binary packages.
+author: '[Dirk Thomas](https://github.com/dirk-thomas)'
+published: true
+categories: Overview
+---
+
+{:toc}
+
+# {{ page.title }}
+
+<div class="abstract" markdown="1">
+{{ page.abstract }}
+</div>
+
+Original Author: {{ page.author }}
+
+## Preface
+
+In the ROS ecosystem a ROS package provides information about its dependencies in the manifest.
+The latest format is specified in the [REP 140](http://www.ros.org/reps/rep-0140.html).
+It supports different kinds of dependencies (`build`, `exec`, `test`, `doc`, etc.) but each dependency is mandatory and needs to satisfied on its own.
+
+This article describes the concept of "dependency groups" which could be used for different purposes.
+In terms of terminology a dependency group is a set of dependencies identified by the name of the group.
+For such a group a different semantic can then be applied (instead of just satisfying each dependency on its own).
+
+## Use Cases
+
+The following subsections will describe how dependencies could be used in two different use cases.
+
+### (1) Need at least one "Provider"
+
+A package might want to declare that at least one dependency from a group needs to be present (called `satisfy-at-least-one`).
```

Comment:
how's the behavior if two packages declare the same dependency group?

Reply:
Each one is evaluated on its own. Not sure I get what you are aiming for with the question?

Reply:
nvmd. I just read the examples which clarified my question.